src/Func.cpp

```diff
@@ -327,8 +339,392 @@ std::string Stage::dump_argument_list() const {
+    intm(init_args) = Tuple(init_vals);
+
+    // Args of the update definition of the intermediate Func
+    vector<Expr> update_args(args.size() + vars_rename.size());
+
+    // We need to substitute the reference to the old RDom's RVars with
+    // the new RDom's RVars. Also, substitute the reference to RVars which
+    // are in 'rvars_kept' with their corresponding new pure Vars.
+    map<string, Expr> substitution_map;
+    for (size_t i = 0; i < intm_rvars.size(); ++i) {
+        substitution_map[intm_rvars[i].var] = intm_rdom[i];
+    }
+    for (size_t i = 0; i < vars_rename.size(); i++) {
+        update_args[i] = vars_rename[i];
+        substitution_map[rvars_kept[i].name()] = update_args[i];
+    }
+    for (size_t i = 0; i < args.size(); i++) {
+        Expr arg = substitute(substitution_map, args[i]);
+        update_args[i + vars_rename.size()] = arg;
+    }
+
+    // Compute the predicates for the intermediate Func and the new update definition.
+    const vector<Expr> predicates = definition.split_predicate();
+    for (const Expr &pred : predicates) {
+        Expr subs_pred = substitute(substitution_map, pred);
+        intm_rdom.where(subs_pred);
+        if (!expr_uses_vars(pred, scope)) {
+            // Only keep the predicate that does not depend on the lifted RVars
+            // (either explicitly or implicitly). For example, if 'rx' is split
+            // into 'rxo' and 'rxi' and 'rxo' is part of the lifted RVars, we'll
+            // ignore every predicate that depends on 'rx'
+            f_rdom.where(pred);
+        }
+    }
+    definition.predicate() = f_rdom.domain().predicate();
+
+    // The update values the intermediate Func should compute
+    vector<Expr> update_vals(values.size());
+    for (size_t i = 0; i < update_vals.size(); i++) {
+        Expr val = substitute(substitution_map, values[i]);
+        // Need to update the self-reference in the update definition to point
+        // to the new intermediate Func
+        val = substitute_self_reference(val, func_name, intm.function(), vars_rename);
+        update_vals[i] = val;
+    }
+    intm(update_args) = Tuple(update_vals);
+
+
+    // Determine the dims and schedule of the update definition of the
+    // intermediate Func. We copy over the schedule from the original
+    // update definition (e.g. split, parallelize, vectorize, etc.).
+    intm.function().update(0).schedule().dims() = dims;
+    intm.function().update(0).schedule().splits() = splits;
+
+    for (size_t i = 0; i < rvars_kept.size(); ++i) {
+        // Apply the purify directive that replaces the RVar in rvars_kept
+        // with a pure Var
+        intm.update(0).purify(rvars_kept[i], vars_rename[i]);
+    }
+
+    // Determine the dims of the new update definition.
+    // Since we're replacing the lifted RVars with Vars in the Func's update
+    // definition, we need to add the pure Vars to the dims list if they are
+    // not already in the list
```

Comment:
bad comment. I was meaning to say the original pure vars not the one from the lifted RVars